### DISCUSSION TITLE
Dynamic reflections and pixi 3d types

### DISCUSSION THREAD

**Discussion by StelyanBaltov** [jnsmalm/pixi3d#96] [General]
Hello, I'm trying to implement dynamic reflections and lighting to a 3d model. Can you please provide, if possible, an example of how to achieve that. Basically, for example, if the 3d object is a shiny car and needs to reflect and be lit up by the sun or some street light or something like that. I see that I can use cubemaps for image based lighting but I'm not sure how to make the dynamic stuff. 

Also some off-topic request, can you please publish pixi3d typings as a package or in the same repo, so that it can be easily used in typescript projects. 

Cheers and have a great day.

**Comment by jnsmalm** [comment]
Hi!

Not exactly sure what you mean with dynamic reflections. Pixi3D supports station reflections with cube maps. If you want dynamic reflections, you probably have to create your own custom render pass, that's a bit too much to describe here. I haven't really tried that myself yet, but it should be possible.

You can find an example at https://codesandbox.io/s/github/jnsmalm/pixi3d-sandbox/tree/master/standard-material?file=/src/index.js If you change the properties "metallic" and "roughness" you will see the difference between a shiny object and a not so shiny object. If you want you can download additional environments from https://ibl.pixi3d.org

The Pixi3D typings are already included in "pixi3d" package from NPM.

**Reply by StelyanBaltov** [reply]
Thank you for quick response as always. By dynamic reflections I mean something like: https://playground.babylonjs.com/#KA93U#243.

**Comment by jnsmalm** [comment]
Sorry, not supported at the moment.

**Reply by StelyanBaltov** [reply]
I will think of something. Cheers mate!
